GearRent/ViewControllers/BookingTableViewCell.m

```diff
@@ -38,13 +34,15 @@ - (void)initializeCell{
     self.statusLabel.text = self.reservation.status;
     PFQuery *listingPhotoQuery = [PFQuery queryWithClassName:@"Listing"];
     [listingPhotoQuery whereKey:@"objectId" equalTo:self.reservation.itemId];
+    __weak typeof(self) weakSelf = self;
     [listingPhotoQuery findObjectsInBackgroundWithBlock:^(NSArray * _Nullable objects, NSError * _Nullable error) {
-        if(error == nil){
+        typeof(self) strongSelf = weakSelf;
+        if(error == nil && strongSelf){
             Item *listing = (Item *) objects[0];
-            self.titleLabel.text = listing.title;
+            strongSelf.titleLabel.text = listing.title;
             PFFileObject *image = (PFFileObject *) listing.images[0];
```

Comment:
Accessing the first element of the array when it is empty can cause a runtime crash.  Prefer to use `array.firstObject` instead of `array[0]` unless you explicitly perform a count check.